modules/issue_tracker/php/edit.class.inc

```diff
@@ -0,0 +1,859 @@
+    }
+
+    /**
+     * Processes the values & saves to database and return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse POST request body.
+        $values = $request->getParsedBody();
+        if (!is_array($values)) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Invalid request'
+            );
+        }
+
+        $user = $request->getAttribute('user');
+        $db   = $this->loris->getDatabaseConnection();
+
+        $issueValues    = [];
+        $validateValues = [];
+        $fields         = [
+            'assignee',
+            'status',
+            'priority',
+            'centerID',
+            'title',
+            'category',
+            'module',
+        ];
+        $fieldsToValidateFirst = [
+            'PSCID',
+            'visitLabel',
+            'centerID',
+        ];
+
+        foreach ($fields as $field) {
+            // The default is a string "null" because if the front end submits
+            // null, it comes through as a string, so the default is to behave
+            // the same way
+            $value = $values[$field] ?? "null";
+            if ($value === "null") {
+                $value = null;
+            }
+            if (isset($field)) {
+                $issueValues[$field] = $value;
+            }
+        }
+        foreach ($fieldsToValidateFirst as $vField) {
+            if (isset($values[$vField]) && $values[$vField] !== "null") {
+                $validateValues[$vField] = $values[$vField];
+            }
+        }
+
+        $issueID = intval($values['issueID']);
+
+        $issueValues['lastUpdatedBy'] = $user->getUsername();
+
+        $validatedInput = $this->validateInput($validateValues, $user);
+        if (array_key_exists('sessionID', $validatedInput)) {
+            $issueValues['sessionID'] = $validatedInput['sessionID'];
+        }
+
+        if (array_key_exists('candID', $validatedInput)) {
+            $issueValues['candID'] = $validatedInput['candID'];
+        }
+
+        // Get changed values to save in history
+        $historyValues = $this->getChangedValues($issueValues, $issueID, $user);
+
+        if (!empty($issueID)) {
+            $db->update('issues', $issueValues, ['issueID' => $issueID]);
+        } else {
+            $issueValues['reporter']    = $user->getUsername();
+            $issueValues['dateCreated'] = date('Y-m-d H:i:s');
+            $db->insert('issues', $issueValues);
+            $issueID = intval($db->getLastInsertId());
+        }
+
+        $this->updateHistory($historyValues, $issueID, $user);
+        $this->updateComments($values['comment'], $issueID, $user);
+
+        // Attachment for new issue.
+        if (isset($_FILES['file'])) {
```

Comment:
There shouldn't be any references to superglobals. This should be using getUploadedFiles from the request: https://www.php-fig.org/psr/psr-7/ (can't directly hyperlink, but you can ctrl-f on the page for the behaviour of getUploadedFiles